crates/network-scheduler/src/data_chunk.rs

```diff
@@ -20,31 +21,40 @@ impl Display for ChunkId {
 
 #[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
 pub struct DataChunk {
-    pub dataset_url: String,
+    pub dataset_id: String,
+    pub download_url: String,
     pub block_range: Range,
     pub size_bytes: u64,
     pub chunk_str: String,
+    pub filenames: Vec<String>,
```

Comment:
I could use `Arc<str>` here, but this struct is serialized and deserialized from the scheduler state, so it would unwrap to the full copies on the next load anyway. If we want to save memory on duplicate strings, we should implement some kind of string pool and use it through the entire code.

Reply:
I guess nevermind then. That's some pretty far-fetched optimization.